client/app.jsx

```diff
@@ -26,24 +43,29 @@ export default class App extends React.Component {
   }
 
   renderPage() {
+    const token = window.localStorage.getItem('userToken');
     const { route } = this.state;
+    let user = null;
+    if (token) {
+      user = jwtDecode(token);
+    }
```

Comment:
Instead of decoding the token on every render, just decode it in handleSignIn / componentDidMount and put your user in App's state.